specification/security/resource-manager/Microsoft.Security/preview/2021-12-01-preview/securityConnectors.json

```diff
@@ -0,0 +1,1055 @@
+          "properties": {
+            "enabled": {
+              "type": "boolean",
+              "description": "Is Microsoft Defender for Endpoint auto provisioning enabled"
+            },
+            "configuration": {
+              "type": "object",
+              "description": "configuration for Microsoft Defender for Endpoint autoprovisioning",
+              "properties": {}
+            }
+          }
+        },
+        "subPlan": {
+          "type": "object",
+          "description": "configuration for the servers offering subPlan",
+          "properties": {
+            "type": {
+              "type": "string",
+              "description": "The available sub plans",
+              "enum": [
+                "P1",
+                "P2"
+              ],
+              "x-ms-enum": {
+                "name": "subPlan",
+                "modelAsString": true
+              }
+            }
+          }
+        }
+      }
+    },
+    "defenderForContainersGcpOffering": {
+      "type": "object",
+      "description": "The containers GCP offering",
+      "x-ms-discriminator-value": "DefenderForContainersGcp",
+      "allOf": [
+        {
+          "$ref": "#/definitions/cloudOffering"
+        }
+      ],
+      "properties": {
+        "nativeCloudConnection": {
+          "type": "object",
+          "description": "The native cloud connection configuration",
+          "properties": {
+            "serviceAccountEmailAddress": {
+              "type": "string",
+              "description": "The service account email address in GCP for this offering"
+            },
+            "workloadIdentityProviderId": {
+              "type": "string",
+              "description": "The GCP workload identity provider id for this offering"
+            }
+          }
+        },
+        "dataPipelineNativeCloudConnection": {
+          "type": "object",
+          "description": "The native cloud connection configuration",
+          "properties": {
+            "serviceAccountEmailAddress": {
+              "type": "string",
+              "description": "The data collection service account email address in GCP for this offering"
+            },
+            "workloadIdentityProviderId": {
+              "type": "string",
+              "description": "The data collection GCP workload identity provider id for this offering"
+            }
+          }
+        },
+        "auditLogsAutoProvisioningFlag": {
+          "type": "boolean",
+          "description": "Is audit logs data collection enabled"
+        },
+        "defenderAgentAutoProvisioningFlag": {
+          "type": "boolean",
+          "description": "Is Microsoft Defender for Cloud Kubernetes agent auto provisioning enabled"
+        },
+        "policyAgentAutoProvisioningFlag": {
+          "type": "boolean",
+          "description": "Is Policy Kubernetes agent auto provisioning enabled"
```

Comment:
consider using an enum instaead as explained here : 

[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html)

Replace boolean/switch properties with better enum

A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].

Reply:
In case of a flag, changing it to enum would not make our life easier here IMO, because even in the case we'll need to add additional values in the future (where enum comes handy), we'll still have to change the property name (having a non-flag property with a "Flag" suffix contradicts its meaning), in any case, this would be a breaking change.